### ISSUE TITLE
Add to solution action ("D396686C-DE0E-4DE6-906D-291CD29FC5DE") fails

### ISSUE THREAD

**Issue by gurustron** [dotnet/templating#2453]
I've stumbled across two scenarios when add to solution action fails. [Repository with reproducers](https://github.com/gurustron/templates-repro).

1. The most critical for me - when project files(`csproj`) are placed to nested folder (not in the same as `.template.config`) replace for `primaryOutputs.path` for `csproj` is not performed:

```
{
  "$schema": "http://json.schemastore.org/template",
  "author": "gurustron",
  "classifications": [
    "MyTestFailsNestedDir"
  ],
  "name": "MyTestFailsNestedDir",
  "identity": "AddFailsNestedDir",
  "shortName": "test-add-sln-nested-fails",
  "tags": {
    "language": "C#"
  },
  "sourceName": "AddFailsNestedDir",
  "preferNameDirectory": true,
  "sources": [
    {
      "source": "./",
      "target": "../"
    }
  ],
  "primaryOutputs": [
    {
      "path": "AddFailsNestedDir.csproj"
    }
  ],
  "postActions": [
    {
      "actionId": "D396686C-DE0E-4DE6-906D-291CD29FC5DE",
      "manualInstructions": [],
      "args": {
        "primaryOutputIndexes": "0"
      }
    }
  ]
}
```

folder strucutre:

```
TemplateRepro.sln\ 
	AddFailsNestedDir\
		.template.config\
			template.json
		AddFailsNestedDir\
			...
			AddFailsNestedDir.csproj
```

2. If `postaction.args.primaryOutputIndexes` is omitted in template.json the replace is not performed and folder is not added to the attempted addition despite next written in the [schema](http://json.schemastore.org/template) "if not specified, all primary outputs are used":

```
{
  "$schema": "http://json.schemastore.org/template",
  "author": "gurustron",
  "classifications": [
    "MyTestFailsOutIndex"
  ],
  "name": "MyTestFailsOutIndex",
  "identity": "MyTestFailsOutIndex",
  "shortName": "test-add-sln-outind-fails",
  "tags": {
    "language": "C#"
  },
  "sourceName": "AddFailsOptIndex",
  "preferNameDirectory": true,
  "primaryOutputs": [
    {
      "path": "AddFailsOptIndex.csproj"
    }
  ],
  "postActions": [
    {
      "actionId": "D396686C-DE0E-4DE6-906D-291CD29FC5DE",
      "manualInstructions": []
    }
  ]
}
```

When project files are on the same level and `postaction.args.primaryOutputIndexes` is specified - everything works:

```
{
  "$schema": "http://json.schemastore.org/template",
  "author": "gurustron",
  "classifications": [
    "MyTestWorks"
  ],
  "name": "MyTestWorks",
  "identity": "AddToSlnWorks",
  "shortName": "test-add-sln-works",
  "tags": {
    "language": "C#"
  },
  "sourceName": "AddToSlnWorks",
  "preferNameDirectory": true,
  "primaryOutputs": [
    {
      "path": "AddToSlnWorks.csproj"
    }
  ],
  "postActions": [
    {
      "actionId": "D396686C-DE0E-4DE6-906D-291CD29FC5DE",
      "manualInstructions": [],
      "args": {
        "primaryOutputIndexes": "0"
      }
    }
  ]
}
``` 
 
I have 5.0 and 3.1.401 sdks installed (using former one via global.json)

**Comment by gurustron** [comment]
@AR-May  Thank you for the info!

**Comment by vlada-shubina** [comment]
Agreed on the following, in case of

following template structure
```
-	.template.config
-	AddFailsNestedDir
-	AddFailsNestedDir\AddFailsNestedDir.csproj
-	AddFailsNestedDir\Program.cs
```

the following template configuration:
```
  "sourceName": "AddFailsNestedDir",
  "preferNameDirectory": true,
  "sources": [
    {
      "source": "./",
      "target": "../"
    }
  ]
```

expected outcome 
``` 
-	<output name>.csproj
-	Program.cs
```

`primaryOutputs` path should contain `AddFailsNestedDir.csproj` as the final location after template instantiation.

The fix should be implemented considering above, the primary outputs returned by API should contain `<output name>.csproj`.
The documentation on `primaryOutputs` to be updated to explain this case.

**Comment by vlada-shubina** [comment]
The issue with incorrect primary outputs in case non-default target or source rename is used is fixed in #2690.
